sdks/python/src/opik/integrations/dspy/callback.py

```diff
@@ -0,0 +1,99 @@
+from typing import Any, Dict, Optional
+
+from dspy.utils.callback import BaseCallback
+
+from opik import context_storage, opik_context
+from opik.api_objects import opik_client, span, trace
+from opik.decorator import error_info_collector
+
+
+class OpikCallback(BaseCallback):
+
+    def __init__(
+        self,
+        project_name: Optional[str] = None,
+    ):
+        self._opik_trace_data: Optional[trace.TraceData] = None
```

Comment:
It limits the callback to process only one span at a time. Storing traces/spans in a dictionary would probably be safer.
We could map call_ids to the corresponding traces and spans and allow people to create a callback they can freely use many times in multiple contexts.